janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/HasStepFolder.java

```diff
@@ -114,16 +113,6 @@ static void foldInIds(final HasStepFolder janusgraphStep, final Traversal.Admin<
                         stepLabels.forEach(janusgraphStep::addLabel);
                         // this has container is no longer needed because its ids will be folded into the JanusGraphStep
                         removableHasContainers.add(hasContainer);
-                        if (!ids.isEmpty()) {
```

Comment:
Yes, this makes sense to remove to me. @sjudeng do you remember why this was necessary previously?

Reply:
@twilmes I think it was another part of the implementation to get one of the TP tests to pass in the 3.2.5 updates and is no longer necessary thanks to your subsequent updates. Glad to see this go away. Thanks @robertdale!